packages/promotion/src/services/promotion-module.ts

```diff
@@ -90,6 +91,120 @@ export default class PromotionModuleService<
     return joinerConfig
   }
 
+  @InjectManager("baseRepository_")
+  async registerUsage(
+    computedActions: PromotionTypes.ComputeActions[],
```

Comment:
```suggestion
    computedActions: PromotionTypes.UsageComputedActions[],
```

Reply:
thought(non-blocking): allow item_id to be undefined here as well